adserver/forms.py

```diff
@@ -60,13 +87,231 @@ class Meta:
+            )
+            self.fields["exclude_countries"].initial = ", ".join(
+                self.instance.excluded_countries
+            )
+            self.fields["include_keywords"].initial = ", ".join(
+                self.instance.included_keywords
+            )
+
+        self.helper = FormHelper()
+
+        self.helper.layout = Layout(
+            Fieldset(
+                "",
+                Div(
+                    Div("start_date", css_class="form-group col-lg-6"),
+                    Div("end_date", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                "live",
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Per impression (CPM) flights"),
+                Div(
+                    Div("cpm", css_class="form-group col-lg-6"),
+                    Div("sold_impressions", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Per click (CPC) flights"),
+                Div(
+                    Div("cpc", css_class="form-group col-lg-6"),
+                    Div("sold_clicks", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Flight targeting"),
+                Div(
+                    "include_countries",
+                    Div(
+                        HTML(
+                            format_html(
+                                """
+                                <ul class="list-inline">
+                                    <li class="list-inline-item"><a class="ea-update-field" href="#" data-value="{}" data-target-field="#id_include_countries">US / Canada</a></li>
+                                    <li class="list-inline-item"><a class="ea-update-field" href="#" data-value="{}" data-target-field="#id_include_countries">EU / AU / NZ</a></li>
+                                    <li class="list-inline-item"><a class="ea-update-field" href="#" data-value="{}" data-target-field="#id_include_countries">APAC</a></li>
+                                    <li class="list-inline-item"><a class="ea-update-field" href="#" data-value="{}" data-target-field="#id_include_countries">Latin America</a></li>
+                                    <li class="list-inline-item"><a class="ea-update-field" href="#" data-value="{}" data-target-field="#id_include_countries">Africa</a></li>
+                                </ul>
+                            """,
+                                ", ".join(us_ca),
+                                ", ".join(eu_aus_nz),
+                                ", ".join(wider_apac),
+                                ", ".join(latin_america),
+                                ", ".join(africa),
```

Comment:
This feels a bit weird -- not really sure waht this is doing.

Reply:
This takes the countries and puts them into `data-value` using `format_html` for safety. Then when somebody clicks the link "US/CA" it populates the relevant countries.

Reply:
Oh I see, they are being passed into a string format?

Reply:
Yes, but passed in in a way that they would be escaped if there was HTML in there.